iml-wire-types/wire-types.rs

```diff
@@ -1235,3 +1235,49 @@ pub mod db {
         }
     }
 }
+
+// Types used for component checks
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ElementState {
+    name: String,
+    configurable: bool,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum ServiceState {
+    Unconfigured,
+    Configured,
```

Comment:
Since the below presumes configured, perhaps it would be better for `Configured` to carry those states internally.

```rust
pub enum RunState {
    Enabled,
    Started,
    Setup
}
```

```rust
pub enum ServiceState {
    Unconfigured,
    Configured(RunState),
}
```



That way, the assumption is made explicit. Also, perhaps `RunState` should match systemd service states?

Reply:
That sounds reasonable to me.